File: client/src/app/_models/summary.ts
import { ShoppingCart } from "./shopping-cart";
import { Voucher } from "./voucher";

export interface Summary{
    AppUserId: number,
    productCost:number,
    discounted: number,
    total: number,
    shoppingCartItems: ShoppingCart[],
    voucherID: number
}

File: API/Entities/Summary.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities
{
    [Table("Summary")]
    public class Summary : BaseEntity
    {   

        [ForeignKey("AppUserId")]
        public int AppUserId{ get; set; }
        public float ProductCost{get;set;} // 208 lei
        public float Discounted{get;set;} // -103 lei
        public float Total{ get; set; }    // 208 - 103 = 105 lei
        public List<ShoppingCart> ShoppingCartItems{get;set;}  

        [ForeignKey("ShoppingCartId")]
        public int ShoppingCartId{get;set;}

        [ForeignKey("VoucherId")]
        public int VoucherId{get;set;}
    }
    
}